apps/backend/src/adoption/repositories/mongodb/adoption-mongo.repository.ts

```diff
@@ -16,39 +19,46 @@ export class AdoptionMongoDBRepository implements AdoptionRepository {
     private readonly adoptionModel: Model<AdoptionDocument>,
   ) {}
 
-  async create(adoption: IAdoption) {
-    const newAdoption = new this.adoptionModel(adoption);
+  async create(adoption: AdoptionEntity) {
+    const newAdoption = new this.adoptionModel(AdoptionMap.toSchema(adoption));
 
     const document = await newAdoption
       .save()
-      .then((adoption) => adoption.populate('user', 'id name'));
+      .then((adoption) =>
+        adoption.populate<PopulatedAdoptionDocument>('user', 'id name'),
+      );
```

Comment:
Does it work without this `then` call? just noticed that the function below doesn't have it

```suggestion
      .populate<PopulatedAdoptionDocument>('user', 'id name')
```

Reply:
Yep, it doest. Good catch!